src/task.c

```diff
@@ -752,39 +770,28 @@ task_system_is_quiescent(task_system *ts)
 
 cleanup:
    for (task_type tt = TASK_TYPE_FIRST; tt < ttlocked; tt++) {
-      rc = platform_condvar_unlock(&ts->group[tt].cv);
+      rc = task_group_unlock(&ts->group[tt]);
       debug_assert(SUCCESS(rc));
    }
 
    return result;
 }
 
-void
+platform_status
 task_perform_until_quiescent(task_system *ts)
 {
+   int wait = 1;
```

Comment:
Thanks for adding some sort of defensive looping mechanism here.

A suggestion: You may want to declare this as `uint64` as that's what `platform_sleep()` requires. And if possible, for readability, rename this as `wait_ns`.